src/Compilers/CSharp/Portable/Syntax/SyntaxNormalizer.cs

```diff
@@ -89,61 +91,66 @@ public override SyntaxToken VisitToken(SyntaxToken token)
+            }
 
-                tk = tk.WithLeadingTrivia(RewriteTrivia(
-                    token.LeadingTrivia,
-                    depth,
-                    isTrailing: false,
-                    indentAfterLineBreak: NeedsIndentAfterLineBreak(token),
-                    mustHaveSeparator: false,
-                    lineBreaksAfter: 0));
+            var tk = token;
 
-                var nextToken = this.GetNextRelevantToken(token);
+            var depth = GetDeclarationDepth(token);
 
-                _afterLineBreak = IsLineBreak(token);
-                _afterIndentation = false;
+            tk = tk.WithLeadingTrivia(RewriteTrivia(
+                token.LeadingTrivia,
+                depth,
+                isTrailing: false,
+                indentAfterLineBreak: NeedsIndentAfterLineBreak(token),
+                mustHaveSeparator: needsSeparatorBefore,
+                lineBreaksBefore: lineBreaksBefore));
 
-                var lineBreaksAfter = LineBreaksAfter(token, nextToken);
-                var needsSeparatorAfter = NeedsSeparator(token, nextToken);
-                tk = tk.WithTrailingTrivia(RewriteTrivia(
-                    token.TrailingTrivia,
-                    depth,
-                    isTrailing: true,
-                    indentAfterLineBreak: false,
-                    mustHaveSeparator: needsSeparatorAfter,
-                    lineBreaksAfter: lineBreaksAfter));
+            _afterLineBreak = IsLineBreak(token);
+            _afterIndentation = false;
 
-                return tk;
-            }
-            finally
+            if (IsRelevant(token))
             {
-                // to help debugging
+                _afterDirective = token.Kind() == SyntaxKind.EndOfDirectiveToken;
                 _previousToken = token;
+                _previousIsInStructuredTrivia = _isInStructuredTrivia;
             }
-        }
 
-        private SyntaxToken GetNextRelevantToken(SyntaxToken token)
-        {
-            // get next token, skipping zero width tokens except for end-of-directive tokens
-            var nextToken = token.GetNextToken(
-                t => SyntaxToken.NonZeroWidth(t) || t.Kind() == SyntaxKind.EndOfDirectiveToken,
-                t => t.Kind() == SyntaxKind.SkippedTokensTrivia);
+            tk = tk.WithTrailingTrivia(RewriteTrivia(
+                token.TrailingTrivia,
+                depth,
+                isTrailing: true,
+                indentAfterLineBreak: false,
+                mustHaveSeparator: false,
+                lineBreaksBefore: 0));
 
-            if (_consideredSpan.Contains(nextToken.FullSpan))
-            {
-                return nextToken;
-            }
-            else
-            {
-                return default(SyntaxToken);
-            }
+            return tk;
         }
 
+        private static bool IsRelevant(SyntaxToken token) => SyntaxToken.NonZeroWidth(token) || token.Kind() == SyntaxKind.EndOfDirectiveToken;
```

Comment:
Previously, what token would be considered "relevant" was impacted by `_consideredSpan`. Now that logic seems to be gone and `_consideredSpan` is unused. Looking at the callers of the first `SyntaxNormalizer` constructor, they were all passing `.FullSpan` anyways.
I think we can remove the `consideredSpan` parameter from that constructor now, and remove the `_consideredSpan` field.